themes/bootstrap3/templates/RecordTab/map.phtml

```diff
@@ -1,14 +1,31 @@
 <? $mapType = $this->tab->getMapType(); ?>
-<? if (isset($mapType) && ($mapType == 'openlayers')) : ?>
-  <? $this->headScript()->appendFile('vendor/ol/ol.js');
-  $this->headScript()->appendFile('map_tab_ol.js');
-  $this->headLink()->appendStylesheet('vendor/ol/ol.css');
-  $mapTabData = $this->tab->getMapTabData();
-  $mapGraticule = $this->tab->getMapGraticule();
-  $basemap = $this->tab->getBasemap();
-  $popupTitle = $this->transEsc('map_results_label');
-  $params = [json_encode($mapTabData), json_encode($popupTitle), json_encode($mapGraticule),
-    json_encode($basemap)];
+<? if (isset($mapType) && (($mapType == 'openlayers') || ($mapType == 'leaflet'))) : ?>
+  <? if ($mapType == 'openlayers') :?>
+     <?
+       $this->headScript()->appendFile('vendor/ol/ol.js');
+       $this->headScript()->appendFile('map_tab_ol.js');
+       $this->headLink()->appendStylesheet('vendor/ol/ol.css');
+     ?>
+  <? elseif ($mapType == 'leaflet') : ?>
+     <?
+       $this->headScript()->appendFile('vendor/leaflet/leaflet.js');
+       $this->headScript()->appendFile('vendor/leaflet/leaflet.latlng-graticule.js');
+       $this->headScript()->appendFile('map_tab_leaflet.js');
+       $this->headLink()->appendStylesheet('vendor/leaflet/leaflet.css');
+     ?>
+  <? endif; ?>
+  <?
+    $mapGraticule = $this->tab->getMapGraticule();
+    $basemap = $this->tab->getBasemap();
+
+    if ($mapType == 'openlayers') {
+      $mapTabData = $this->tab->getMapTabData();
+      $popupTitle = $this->transEsc('map_results_label');
+      $params = [json_encode($mapTabData), json_encode($popupTitle), json_encode($mapGraticule), json_encode($basemap)];
+    } elseif ($mapType == 'leaflet') {
+      $mapTabData = $this->tab->getLeafletMapTabData();
```

Comment:
If you're always setting a $mapTabData variable in the template, I wonder if it would be cleaner to make $this->tab->getMapTabData smarter internally, so it just returns the right kind of data, rather than having to create a separate public getLeafletMapTabData method. That way, if in future you remove the OpenLayers implementation, there's not a Leaflet-specific method name floating around as a potential source of confusion.

(This is a very minor point, by the way -- just something that struck me as a potential code clarification).

Reply:
Good point. And it would give us the flexibility to change mapping platforms in the future if something better than leaflet comes along.